app/javascript/src/ActiveDocs/components/ActiveDocsSpec.jsx

```diff
@@ -1,18 +1,123 @@
+const AUTOCOMPLETE_PARAMS_MAP = {
+  'app_key': 'app_keys',
+  'app_id': 'app_ids',
+  'application_id': 'application_ids',
+  'user_key': 'user_keys',
+  'user_id': 'user_ids',
+  'account_id': 'account_ids',
+  'metric_name': 'metric_names',
+  'metric_id': 'metric_ids',
+  'backend_api_metric_name': 'backend_api_metric_names',
+  'service_id': 'service_ids',
+  'admin_id': 'admin_ids',
+  'service_plan_id': 'service_plan_ids',
+  'application_plan_id': 'application_plan_ids',
+  'account_plan_id': 'account_plan_ids',
+  'client_id': 'client_ids',
+  'client_secret': 'client_secrets',
+  'service_token': 'service_tokens',
+  'access_token': 'access_token'
+}
+
+type Primitive = string | number | boolean
+
+type AccountData = {
+  [string]: Array<{name: string, value: string}> | []
+}
+
+type SwaggerResponse = {
+  body: {
+    [string]: mixed
+  },
+  data: string,
+  headers: {},
+  obj: {},
+  ok: boolean,
+  status: number,
+  statusText: string,
+  text: string,
   url: string
 }
 
-const ActiveDocsSpec = ({ url }: ActiveDocsSpecProps) => <SwaggerUI url={url} />
+type ActiveDocsSpecProps = {
+  accountData?: AccountData,
+  url: string,
+}
+
+type ParamArraySchema = {
+  type: 'array',
+  items: { type: Primitive }
+}
+type ParamEnumSchema = {
+  enum: Array<Primitive>,
+  items: { type: Primitive }
+}
+
+type Examples = Array<{
+  summary: string,
+  value: string
+}>
+
+type Param = {
+  in: string,
+  name: string,
+  description?: string,
+  required?: boolean,
+  schema?: Primitive | ParamArraySchema | ParamEnumSchema,
+  examples?: Examples
+}
+
+const addAutocompleteToParam = (param: Param, accountData: AccountData): Param => {
+  const paramName = AUTOCOMPLETE_PARAMS_MAP[param.name]
+  const autocompleteData = accountData[paramName]
+  const paramHasAutocompleteData = autocompleteData.length > 0 &&
+    autocompleteData.every(param => param.value !== '')
+
+  return paramHasAutocompleteData
+    ? {
+      ...param,
+      examples: autocompleteData.reduce((examples, item) => (
+        [...examples, {summary: item.name, value: item.value}]
+      ), [{ summary: 'Select an option to autocomplete', value: '-' }])
+    }
+    : param
+}
+
+const injectAccountDataToResponse = (response: SwaggerResponse, accountData: AccountData): SwaggerResponse => {
+  const responseWithAutocomplete = { ...response }
+  const paths = { ...response.body.paths }
+  let pathsWithAutocomplete = {}
+
+  for (let key in paths) {
+    pathsWithAutocomplete[key] = paths[key]
+    const parametersWithAutocomplete = paths[key].parameters
+      .map(param => addAutocompleteToParam(param, accountData))
+    pathsWithAutocomplete[key].parameters = parametersWithAutocomplete
+  }
+  responseWithAutocomplete.body.paths = pathsWithAutocomplete
+  responseWithAutocomplete.data = JSON.stringify(responseWithAutocomplete.body)
+  responseWithAutocomplete.text = JSON.stringify(responseWithAutocomplete.body)
+  return responseWithAutocomplete
+}
```

Comment:
You could do something similar to what I suggested in `addAutocompleteToParam` here.